### PR TITLE
FILE-6 - Add more available date ranges to the date range dropdown

### PR DESCRIPTION
**Pull request by gitstart**
#### Add more available date ranges to dropdown
- [x] Past N amount of days
- [x] Customizable date range
- [x] All monthly earnings periods from November 2022 to the current month

## Issue Reference

https://github.com/filecoin-saturn/L1-dashboard/issues/14

## Demo
https://www.loom.com/share/db7582115b7e4d09857ec97f7dcbcd94?sid=2dc1bf95-af5a-4576-82d2-214ed2b9a268

> This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### CODE REVIEW

**Comment by gitstart** [comment]
@DiegoRBaquero this PR is ready for review

**Comment by gitstart** [comment]
@guanzo @joaosa @DiegoRBaquero this PR is ready for review

**Review by DiegoRBaquero — APPROVED** [review]
lgtm but want a second approver

File: src/date-utils.ts [hunk 1]
```diff
@@ -53,3 +54,48 @@ export function dayIntervalFromDateRange(startDate: Date, endDate: Date) {
...
+  }
+
+  return options.reverse();
+}
+
+export function parseDateRange(dateString: string): {
+  startDate: Date;
+  endDate: Date;
+} | null {
+  const customRangeMatch = /^(\d{4})-(\d{2})-(\d{2}) (\d{4})-(\d{2})-(\d{2})$/g.exec(dateString);
```
Comment (AmeanAsad) [inline src/date-utils.ts:87]: Can you add a comment explaining what this regex does exactly?

File: src/pages/dashboard/index.tsx [hunk 2]
```diff
@@ -19,51 +27,80 @@ interface OverviewProps {
...
+      };
       step = "day";
       labelUnit = "day";
       break;
-    case TimePeriod.MONTH:
-      dateRange = pastDateRange("day", 30);
+    }
+    case FilterType.DateRange: {
+      if (periodRange.dateRange === undefined || periodRange.dateRange.startDate > periodRange.dateRange.endDate) {
+        throw new Error("Invalid date range");
```
Comment (AmeanAsad) [inline src/pages/dashboard/index.tsx:95]: Great that you're adding all this error handling in the functions. However, not seeing those errors being handled on the front end when you call this function.

**Pull request closed without merging**